### PR TITLE
[opentelemetry] Allow to specify collector image

### PR DESCRIPTION
**Pull request by songy23**
## Description

Allow to specify OTel collector image for CI usage

## Reviewer checklist

* [ ] If this PR modifies anything else than strictly the default scenario, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/system-tests-ci.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

### CODE REVIEW

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
It's failing : https://github.com/DataDog/system-tests/actions/runs/5412365211/jobs/9836381313?pr=1318

**Comment by songy23** [comment]
Just back from vacation, I'm taking a look today

**Comment by songy23** [comment]
Should be fixed now https://github.com/DataDog/system-tests/actions/runs/5532265026/jobs/10094125917?pr=1318, PTAL @cbeauchesne

File: utils/_context/containers.py [hunk 1]
```diff
@@ -516,9 +516,10 @@ def __init__(self, host_log_folder) -> None:
 
 class OpenTelemetryCollectorContainer(TestedContainer):
     def __init__(self, host_log_folder) -> None:
+        image = os.environ.get("OTEL_COLLECTOR_IMAGE", "otel/opentelemetry-collector-contrib:latest")
```
Comment (cbeauchesne) [inline utils/_context/containers.py:519]: Could you just change then env var name to `SYSTEM_TESTS_OTEL_COLLECTOR_IMAGE` ? I'm expecting soon or later some trouble with env var name conlfict ...

```suggestion
        image = os.environ.get("SYSTEM_TESTS_OTEL_COLLECTOR_IMAGE", "otel/opentelemetry-collector-contrib:latest")
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: utils/_context/containers.py (hunk 1 above)
Comment (songy23) [reply utils/_context/containers.py:519]: SGTM

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**